assets/src/hooks/useDetour.ts

```diff
@@ -27,37 +28,74 @@ export enum DetourState {
   Finished,
 }
 
-export const useDetour = ({ routePatternId, shape }: OriginalRoute) => {
-  const [snapshot, send] = useMachine(createDetourMachine)
+const selectWaypoints = (snapshot: SnapshotFrom<typeof createDetourMachine>) =>
```

Comment:
Since waypoints in one context != waypoints in another context (as in, the snapshot includes start and end points as waypoints) a comment here would be helpful as a waypoint ;)

Reply:
I'm considering renaming this but I'm struggling to find names. I was thinking maybe "selectDetourWaypoints"/"selectDetourRouteWaypoints"/"selectRouteWaypoints" but I don't think it makes it any clearer?

Maybe `intermediateWaypoints`? Really not sure, I'll try to come up with a comment to describe this at the very least

Reply:
Maybe this is more of a UI concern and `useDetour` shouldn't be computing this? (something to revisit later I suppose)

Reply:
https://github.com/mbta/skate/compare/068d53704c5f20a0db8a02955c1161d454c7c612..def441d30e8274e09b9b1b6e6f86fc7e325c3502